### PR TITLE
Create Header: fix Pegasus display name in sign-in dropdown

### PR DESCRIPTION
**Pull request by Erin007**
Follow up to #25004 
Currently, on Pegasus pages the new sign in icon dropdown is showing a generic "Student" or "Teacher" rather than the user's display name.

BEFORE:
screen shot 2018-10-06 at 6 16 52 pm
As discussed [here](https://github.com/code-dot-org/code-dot-org/pull/25004#discussion_r220053372) 

This is because there is some wizardry to retrieving the user's display name `current_user.short_name` that was relying on being in a `.user_name` div.  However, I want that wizardry to also apply to `.display_name`.

AFTER: 
screen shot 2018-10-06 at 7 55 29 pm

I want to keep that div named `.display_name` for two reasons: 1.) I don't want the styling to interfere with `.user_name` because this dropdown item is styled differently than they other options in the dropdown. 2.) `.display_name` is more accurate because we are actually showing the user's display name, not their username. 
screen shot 2018-10-06 at 7 55 47 pm

### CODE REVIEW

File: shared/haml/user_header.haml [hunk 1]
```diff
@@ -107,6 +107,17 @@
...
+    var displayNameSpan = document.querySelector('.header_button.header_user.user_menu .display_name');
+    if (displayNameSpan) {
+      var dataName = displayNameSpan.dataset.shortname;
+      var id = displayNameSpan.dataset.id;
+      var cookieMatch = document.cookie.match('(^|;) ?' + cookieKey + '=([^;]*)(;|$)');
+      var storedName = cookieMatch && cookieMatch[2];
+      if (storedName && !id) {
+        displayNameSpan.innerHTML = displayNameSpan.innerHTML.replace(dataName, storedName);
+      }
+    }
```
Comment (breville) [inline shared/haml/user_header.haml:113]: it feels like we could factor out some code between this block and the one above it?

Comment (Erin007) [reply shared/haml/user_header.haml:113]: I extracted the two shared lines. Is that what you were thinking?

Comment (breville) [reply shared/haml/user_header.haml:113]: Just by glancing, it looks as though there is a bigger possible duplication to factor out; we seem to have basically the same chunk of code which could be parameterised with `user_name` vs. `display_name` and `nameSpan` vs. `displayNameSpan`.

Comment (Erin007) [reply shared/haml/user_header.haml:113]: Like this?

Comment (breville) [reply shared/haml/user_header.haml:113]: That looks great.  One suggestion is that "get" functions normally return a value, so maybe a different name to indicate that it actually does the work.

File: shared/haml/user_header.haml [hunk 2]
```diff
@@ -91,18 +91,18 @@
...
     // here we should be sure to also update other locations.
     var nameSpan = document.querySelector('.header_button.header_user.user_menu .user_name');
     var displayNameSpan = document.querySelector('.header_button.header_user.user_menu .display_name');
 
-    function getUserShortName(element) {
+    function retrieveUserShortName(element) {
       if (element) {
+        var cookieMeta = "path=/;domain=" + topLevelDomain;
+        var cookieMatch = document.cookie.match('(^|;) ?' + cookieKey + '=([^;]*)(;|$)');
+        var storedName = cookieMatch && cookieMatch[2];
```
Comment (breville) [inline shared/haml/user_header.haml:106]: Nice moving these too!

**Pull request merged**